common-react/@dbeaver/ui-kit/src/Button/ButtonRoot.tsx

```diff
@@ -0,0 +1,21 @@
+import { Button as AriaButton, type ButtonProps as AriaKitButtonProps } from '@ariakit/react';
+import { Spinner } from '../Spinner/Spinner.js';
+
+export interface ButtonPropsWithLoader extends Omit<AriaKitButtonProps, 'clickOnEnter' | 'clickOnSpace'> {
+  loading?: boolean;
+  loader?: React.ReactNode;
+}
+
+export function ButtonRoot({ loading, loader, children, onClick, ...props }: ButtonPropsWithLoader) {
+  if (loading) {
+    props['aria-busy'] = true;
+    props['data-loading'] = true;
+  }
+
+  return (
+    <AriaButton onClick={loading ? () => null : onClick} {...props}>
+      {loading && (loader ? loader : <Spinner className="tw:absolute tw:z-1" />)}
+      {children}
```

Comment:
Usually, buttons in the loading state hide their content to avoid visual collisions